aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -27,37 +27,192 @@ Kestrel supports the following features:
+* [ListenOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/ListenOptions.cs)
+* [KestrelServerLimits](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs)
+
+
+---
+
+### Kestrel options
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+For information about Kestrel options, see [KestrelServerOptions class](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.server.kestrel.kestrelserveroptions).
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+The Kestrel web server has constraint configuration options that are especially useful in Internet-facing deployments. Here are some of the limits you can set:
+
+- Maximum client connections
+- Maximum request body size
+- Minimum request body data rate
+
+You set these constraints and others in the The `Limits` property of the [KestrelServerOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerOptions.cs) class. The `Limits` property holds an instance of the [KestrelServerLimits](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs) class. 
+
+**Maximum client connections**
+
+The maximum number of concurrent open HTTP/HTTPS connections can be set for the entire application with the following code:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=3-4)]
+
+There's a separate limit for connections that have been upgraded from HTTP or HTTPS to another protocol (for example, on a WebSockets request).  After a connection is upgraded, it’s not counted against the `MaxConcurrentConnections` limit anymore.
+
+**Maximum request body size**
+
+The default maximum request body size is 30,000,000 bytes, which is approximately 28.6MB. Here's an example that shows how to configure the constraint for the entire application:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=5)]
+
+This will affect every request.  The recommended way to override the limit in an ASP.NET Core MVC app is to use the [RequestSizeLimit](https://github.com/aspnet/Mvc/blob/rel/2.0.0/src/Microsoft.AspNetCore.Mvc.Core/RequestSizeLimitAttribute.cs) attribute on an action method:
+
+```csharp
+[RequestSizeLimit(100000000)]
+public IActionResult MyActionMethod()
+```
+
+You can also override the setting on a specific request as shown here:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_Limits&highlight=3-4)]
+ 
+An exception is thrown if you try to configure the limit on a request after the application has started reading the request. There’s an `IsReadOnly` property that tells you if the request body is in read-only state, meaning it’s too late to configure the limit.
```

Comment:
* "There's" has another smart (curly) apostrophe. ..... and another one down on "it's"
* "application" :point_right: "app"